### PR TITLE
Documentation: Request

### LINKED ISSUE #652: Update documentation for Request
**What is not easy to do right now?**
The doc is outdated in https://github.com/dream11/zio-http/blob/main/docs/index.md#accessing-the-request

**Describe the solution you'd like**
Update the code snippets as per RC18

**Why is it important?**
Docs and examples help developers understand how to use zhttp.

### PR DESCRIPTION
**Pull request by ShrutiVerma97**

### CODE REVIEW

File: docs/website/docs/v1.x/dsl/request/index.md [hunk 1]
```diff
@@ -1 +1,58 @@
...
+`Request` can be created with `method`, `url`, `headers`, `remoteAddress` and `data`.
+
+The below snippet creates a request with default params, `method` as `Method.GET`, `url` as `URL.root`, `headers` as `Headers.empty`, `data` as `HttpData.Empty`, `remoteAddress` as `None`
+```scala
+val request: Request = Request()
+```
+ ## Creating a Request Effectfully
+ You can create request effectfully using `make` constructor in `Request`.
+ 
+ The below snippet creates an effectful request with default params, method as `Method.GET`, url as `URL.root`, headers as `Headers.empty`, data as `HttpData.Empty`, remoteAddress as `None`
```
Comment (amitksingh1490) [inline docs/website/docs/v1.x/dsl/request/index.md]: This Part I believe will never be used on Server- Side apart from unit test and for Client side we have `clientParams` lets discuss this

**Comment by amitksingh1490** [comment]
Refer headers section in doc and structure this accordingly

File: docs/website/docs/v1.x/dsl/request/index.md [hunk 2]
```diff
@@ -1 +1,86 @@
...
+`Request` can be created with `method`, `url`, `headers`, `remoteAddress` and `data`.
+
+The below snippet creates a request with default params, `method` as `Method.GET`, `url` as `URL.root`, `headers` as `Headers.empty`, `data` as `HttpData.Empty`, `remoteAddress` as `None`
+```scala
+val request: Request = Request()
+```
+
+## Using Request in creating HTTP apps
+
+You can create an HTTP app which accepts a request of type `Request` and produces response of type `Response`:
```
Comment (amitksingh1490) [inline docs/website/docs/v1.x/dsl/request/index.md]:
```suggestion
You can create an HTTP app which accepts a input of type `Request` and produces output of type `Response`:
```

File: docs/website/docs/v1.x/dsl/request/index.md [hunk 3]
```diff
@@ -1 +1,86 @@
-# Work in progress
\ No newline at end of file
+# Request
+ 
+## Server-side
+
+`Request` is designed in **ZIO HTTP** in the simplest way possible to make HTTP calls.
+ It supports all HTTP request methods (as defined in [RFC2616](https://datatracker.ietf.org/doc/html/rfc2616) ) and headers along with custom methods and headers.
```
Comment (amitksingh1490) [inline docs/website/docs/v1.x/dsl/request/index.md]:
```suggestion
**ZIO HTTP** `Request` is designed in the simplest way possible to, decode HTTP `Request` into a ZIO HTTP requests.
 It supports all HTTP request methods (as defined in [RFC2616](https://datatracker.ietf.org/doc/html/rfc2616) ) and headers along with custom methods and headers.
```

File: docs/website/docs/v1.x/dsl/request/index.md [hunk 4]
```diff
@@ -1 +1,101 @@
...
+According to the request path, it will respond with the corresponding response:
+- if the request has path `/name` it will match the first route.
+- if the request has path `/name/joe` it will match the second route as `/:` matches the path partially as well.  
+
+ ```scala
+ val app: HttpApp[Any, Nothing] = Http.collect[Request] {
+    case Method.GET -> !! / a => Response.text(s"$a")
+    case Method.GET -> !! /: rest => Response.text(s"$rest")
+  }
+```
```
Comment (amitksingh1490) [inline docs/website/docs/v1.x/dsl/request/index.md]:
```suggestion
 ```scala
 val app: HttpApp[Any, Nothing] = Http.collect[Request] {
    case Method.GET -> !! / a => Response.text(s"$a")
    case Method.GET -> !! /a/: rest => Response.text(s"$rest")
  }
```
```

File: docs/website/docs/v1.x/dsl/request/index.md [hunk 5]
```diff
@@ -1 +1,101 @@
...
+
+#### Matching path on the basis of Type
+
+The below snippet will match path like `/fruits/3/apples` but not `/fruits/a/apples`
+
+```scala
+val app: HttpApp[Any, Nothing] = Method.GET / "fruits" / *[Int] / "apples" to { a =>
+    Response.text(s"apples: ${a.params.toString}")
+  }
+```
```
Comment (amitksingh1490) [inline docs/website/docs/v1.x/dsl/request/index.md]: remove this section its not stable and doesn't support queryparams as of now

File: docs/website/docs/v1.x/dsl/request/index.md [hunk 6]
```diff
@@ -1 +1,101 @@
...
+
+```scala
+val app: HttpApp[Any, Nothing] = Method.GET / "fruits" / *[Int] / "apples" to { a =>
+    Response.text(s"apples: ${a.params.toString}")
+  }
+``` 
+
+## Accessing the Request
+
+- `getBody` to access the content of request as a Chunk of Bytes
```
Comment (amitksingh1490) [inline docs/website/docs/v1.x/dsl/request/index.md]:
```suggestion
- `getBody` to access the content of request as a Chunk[Byte]
```

File: docs/website/docs/v1.x/dsl/request/index.md [hunk 7]
```diff
@@ -1 +1,91 @@
-# Work in progress
\ No newline at end of file
+# Request
+ 
+**ZIO HTTP** `Request` is designed in the simplest way possible to, decode HTTP `Request` into a ZIO HTTP request.
```
Comment (amitksingh1490) [inline docs/website/docs/v1.x/dsl/request/index.md]:
```suggestion
**ZIO HTTP** `Request` is designed in the simplest way possible to decode HTTP Request into a ZIO HTTP request.
```

File: docs/website/docs/v1.x/dsl/request/index.md [hunk 8]
```diff
@@ -1 +1,91 @@
...
+Creating requests using `Request` is useful while writing unit tests.
+
+The below snippet creates a request with default params, `method` as `Method.GET`, `url` as `URL.root`, `headers` as `Headers.empty`, `data` as `HttpData.Empty`, `remoteAddress` as `None`
+```scala
+val request: Request = Request()
+```
+
+## Matching and Extracting Requests
+
+`Request` can be extracted into an HTTP Method and Path via the -> object. On the left side is the `Method`, and on the right side, the `Path`.
```
Comment (sumawa) [inline docs/website/docs/v1.x/dsl/request/index.md]: ```and Path via the -> object```
you can omit "object" or may be use some other term?

File: docs/website/docs/v1.x/dsl/request/index.md [hunk 9]
```diff
@@ -1 +1,91 @@
...
+### Method
+ `Method` represents HTTP methods like POST, GET, PUT, PATCH, and DELETE.
+You can create existing HTTP methods such as `Method.GET`, `Method.POST` etc or create a custom one.
+ 
+
+### Path
+ `Path` can be created using
+  - `!!` which represents the root
+  - `/` which represents the path delimiter and starts the extraction from the left-hand side of the expression
+  - `/:` which represents the path delimiter and starts the extraction from the right-hand side of the expression and can match paths partially
```
Comment (sumawa) [inline docs/website/docs/v1.x/dsl/request/index.md:32]: Can there be a small example here explaining the term "extraction from left or right side"?

Comment (ShrutiVerma97) [reply docs/website/docs/v1.x/dsl/request/index.md:32]: I added just below this

File: docs/website/docs/v1.x/dsl/request/index.md [hunk 10]
```diff
@@ -1 +1,91 @@
...
+Creating requests using `Request` is useful while writing unit tests.
+
+The below snippet creates a request with default params, `method` as `Method.GET`, `url` as `URL.root`, `headers` as `Headers.empty`, `data` as `HttpData.Empty`, `remoteAddress` as `None`
+```scala
+val request: Request = Request()
+```
+
+## Matching and Extracting Requests
+
+`Request` can be extracted into an HTTP Method and Path via the ->. On the left side is the `Method`, and on the right side, the `Path`.
```
Comment (amitksingh1490) [inline docs/website/docs/v1.x/dsl/request/index.md]:
```suggestion
`Request` can be extracted into an HTTP Method and Path via `->`. On the left side is the `Method`, and on the right side, the `Path`.
```

**Review by amitksingh1490 — APPROVED** [review]

**Pull request merged**